src/Workleap.Extensions.MediatR.Tests/PreAndPostProcessorTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using GSoft.Extensions.Xunit;
+using MediatR;
+using MediatR.Pipeline;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Workleap.Extensions.MediatR.Tests;
+
+public sealed class PreAndPostProcessorTests : BaseUnitTest<PreAndPostProcessorTests.Fixture>
+{
+    public PreAndPostProcessorTests(Fixture fixture, ITestOutputHelper testOutputHelper)
+        : base(fixture, testOutputHelper)
+    {
+    }
+
+    [Fact]
+    public async Task SendAsync_Executes_Automatically_Registered_Pre_And_Post_Processors()
+    {
+        var mediator = this.Services.GetRequiredService<IMediator>();
+        var state = this.Services.GetRequiredService<TestState>();
+
+        await mediator.SendAsync(new MyCommand(), CancellationToken.None);
+        await mediator.SendAsync(new MyQuery(), CancellationToken.None);
+
+        Assert.Equal(1, state.MyHandler_MyCommand_CallCount);
+        Assert.Equal(1, state.MyHandler_MyQuery_CallCount);
+
+        Assert.Equal(1, state.MyCommandPreProcessor_Pre_MyCommand_CallCount);
+        Assert.Equal(1, state.MyCommandPostProcessor_Post_MyCommand_CallCount);
+
+        Assert.Equal(1, state.MyFirstQueryPreAndPostProcessor_Pre_MyQuery_CallCount);
+        Assert.Equal(1, state.MyFirstQueryPreAndPostProcessor_Post_MyQuery_CallCount);
+
+        Assert.Equal(1, state.MySecondQueryPreAndPostProcessor_Pre_MyQuery_CallCount);
+        Assert.Equal(1, state.MySecondQueryPreAndPostProcessor_Post_MyQuery_CallCount);
```

Comment:
We basically validate that pre & post processors are automatically registered as it should be (before MediatR intrduced the breaking change, then the bug).

If MediatR introduces the bug again in the future, we'll know.